packages/vitest/src/node/core.ts

```diff
@@ -447,51 +450,61 @@ export class Vitest {
     await this.report('onInit', this)
     await this.report('onPathsCollected', files.flatMap(f => f.filepath))
 
-    const workspaceSpecs = new Map<TestProject, File[]>()
+    const specifications: TestSpecification[] = []
     for (const file of files) {
       const project = this.getProjectByName(file.projectName || '')
-      const specs = workspaceSpecs.get(project) || []
-      specs.push(file)
-      workspaceSpecs.set(project, specs)
+      const specification = project.createSpecification(file.filepath, undefined, file.pool)
+      specifications.push(specification)
     }
 
-    for (const [project, files] of workspaceSpecs) {
-      const filepaths = files.map(f => f.filepath)
-      this.state.clearFiles(project, filepaths)
-      files.forEach((file) => {
-        file.logs?.forEach(log => this.state.updateUserLog(log))
-      })
-      this.state.collectFiles(project, files)
-    }
-
-    await this.report('onCollected', files).catch(noop)
+    await this.report('onSpecsCollected', specifications.map(spec => spec.toJSON()))
+    await this._testRun.start(specifications).catch(noop)
 
     for (const file of files) {
+      const project = this.getProjectByName(file.projectName || '')
+      await this._testRun.enqueued(project, file).catch(noop)
+      await this._testRun.collected(project, [file]).catch(noop)
+
       const logs: UserConsoleLog[] = []
       const taskPacks: TaskResultPack[] = []
 
       const tasks = getTasks(file)
+      const events: TaskEventPack[] = [
+        [file.id, 'suite-prepare'],
+      ]
       for (const task of tasks) {
         if (task.logs) {
           logs.push(...task.logs)
         }
-        taskPacks.push([task.id, task.result, task.meta])
+        if (task.type === 'test') {
+          taskPacks.push([task.id, task.result, task.meta])
+          if (task.mode !== 'skip' && task.mode !== 'todo') {
+            events.push([task.id, 'test-prepare'], [task.id, 'test-finished'])
+          }
+        }
```

Comment:
Is ignoring skipped tests here intentional?

Reply:
Of course. They are ignored by the runner, we report them at the end of suite: https://github.com/vitest-dev/vitest/pull/7069/files#diff-00f8b4e68ee1b24f202a98a83759158b273f298a31033bad9ac50e8ea887055fR97

Reply:
The suite skip on the next line seems unintentional though